src/vos/vos_space.c

```diff
@@ -372,3 +375,50 @@ vos_space_unhold(struct vos_pool *pool, daos_size_t *space_hld)
 	POOL_SCM_HELD(pool)	-= space_hld[DAOS_MEDIA_SCM];
 	POOL_NVME_HELD(pool)	-= space_hld[DAOS_MEDIA_NVME];
 }
+
+void
+vos_space_update_metrics(struct vos_pool *pool)
+{
+	struct vos_pool_metrics	*vpm;
+	uint64_t		 now;
+
+	vpm = pool->vp_metrics;
+	if (!vpm)
+		return;
+
+	now = daos_gettime_coarse();
+	if (now < vpm->vp_space_metrics.vsm_last_update_ts + VOS_SPACE_METRICS_INTV) {
```

Comment:
This is not necessary, vos_gc_pool() is executed in 10s interval when there is no work for GC (see gc_ult()), so this vos_space_update_metrics() could be delayed 20 seconds after this check.

The vos_space_update_metrcis() isn't a heavy function, we can afford calling it every time in vos_gc_pool().

Reply:
Fixed with commit 29f2ddcecc7a7b9f630cb283f29484715580e7a8